File: runtime/src/de/fatox/meta/injection/Named.java
package de.fatox.meta.injection;

import java.lang.annotation.Retention;
import java.lang.annotation.Documented;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * String-based {@linkplain Qualifier qualifier}.
 *
 * <p>Example usage:
 *
 * <pre>
 *   public class Car {
 *     &#064;Inject <b>@Named("driver")</b> Seat driverSeat;
 *     &#064;Inject <b>@Named("passenger")</b> Seat passengerSeat;
 *     ...
 *   }</pre>
 */
@Qualifier
@Documented
@Retention(RUNTIME)
public @interface Named {

    /** The name. */
    String value() default "";
}

File: runtime/src/main/kotlin/de/fatox/meta/injection/Named.kt
package de.fatox.meta.injection

import java.lang.annotation.Documented
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * String-based [qualifier][Qualifier].
 *
 *
 * Example usage:
 *
 * <pre>
 * public class Car {
 * &#064;Inject **@Named("driver")** Seat driverSeat;
 * &#064;Inject **@Named("passenger")** Seat passengerSeat;
 * ...
 * }</pre>
 */
@Qualifier
@Documented
@Retention(RetentionPolicy.RUNTIME)
annotation class Named(
    /** The name.  */
    val value: String = ""
)